api/services/SiteCreator.js

```diff
@@ -1,152 +1,184 @@
-const GitHub = require("./GitHub")
```

Comment:
Not a blocker, but I think there's a good opportunity to split this out into 2 services. One for creating sites and one for adding a user to a site.

Reply:
yeahhhhhhh....that whole SiteCreator thing is pretty huge. I was trying to not mess with it as much as possible since it is pretty intricate. Could definitely stand for some refactoring.

Reply:
I actually don't see where we render an error if a site already exists?

Reply:
If a site already exists, the user gets added to the site (_if_ they have the valid github permissions). It is the same behavior that we already had.

Reply:
If they don't have the proper GitHub permissions, then then the error alert gets shown on the Site List page that the user is redirected to (same behavior as before).

Reply:
Ah got it, so the site fails to validate because it's missing the `engine` and `defaultBranch` props. That could get sticky even if we check the status code since a validation error on `owner` or `repo` would cause these fields to be displayed 🤔

Reply:
We do have some frontend validations on those though... Maybe something we could tie up if we come back after this PR and refactor the SiteCreator?

Reply:
I'm not quite following.
When just the owner+repo fields are shown, the submission triggers the "add user to site" flow. If that fails (because the site doesn't exist in Federalist), then the form changes into the regular Add New Site form (with the owner, repo, engine, branch fields), and submitting it triggers the regular "add site" flow.

Reply:
Okay, I've been thinking about it backwards. I see now. I think we're good as long as we check for a 404. I still think there's a chance for a permissions error to get swallowed if we don't.

Reply:
...since `.addUserToSite` calls `. checkExistingGithubRepository` which will raise a permissions error and pass that down to the frontend.

Reply:
yeah, you're right